### PR TITLE
Payment-specific discounts

### PR DESCRIPTION
**Pull request by shahthepro**
- [x] Add payment-specific discounts
- [x] Add `minCartValue` and `maxDiscountValues`

### CODE REVIEW

**Comment by shahthepro** [comment]
@nick @franckc The PR is ready to be reviewed. But I'll have to do some more tests before and after merging/deploying to prod.

File: backend/models/discount.js [hunk 1]
```diff
@@ -34,6 +38,15 @@ module.exports = (sequelize, DataTypes) => {
       },
       uses: {
         type: DataTypes.INTEGER
+      },
+      data: {
+        type: isPostgres ? DataTypes.JSONB : DataTypes.JSON
+      },
+      minCartValue: {
```
Comment (franckc) [inline backend/models/discount.js:45]: nit: maybe add a comment explaining in what unit this value is. Is it in cents? Same thing for maxDiscountValue below

**Review by franckc — APPROVED** [review]
Overall LGTM. Nice work on this!

Before you merge, could you make sure to:
1. rebase off master. I did some refactoring/bug fixing of the Discount code that will impact this PR.
2. add some back-end unit test to ensure the payment specific discount work as expected (I added new tests for percentage and fixed discount).

File: backend/models/discount.js (hunk 1 above)
Comment (shahthepro) [reply backend/models/discount.js:45]: Yep, it's in cents. Will add a comment

**Pull request merged**